### PR TITLE
repart: use correct errno

### PR DESCRIPTION
**Pull request by YHNdnzj**

### CODE REVIEW

**Comment by DaanDeMeyer** [comment]
Ouch thanks!

**Review by DaanDeMeyer — APPROVED** [review]

**Review by bluca — APPROVED** [review]

**Comment by bluca** [comment]
Is this a regression in 255?

**Review by yuwata — APPROVED** [review]

File: src/partition/repart.c [hunk 1]
```diff
@@ -2331,7 +2331,7 @@ static int context_load_partition_table(Context *context) {
                         return r;
 
                 if (fstat(context->backing_fd, &st) < 0)
-                        return log_error_errno(r, "Failed to stat %s: %m", context->node);
+                        return log_error_errno(errno, "Failed to stat %s: %m", context->node);
```
Comment (yuwata) [inline src/partition/repart.c:2334]: At least, this is a regression in v255 (93f125a66f4).

File: src/partition/repart.c [hunk 2]
```diff
@@ -6359,7 +6359,7 @@ static int context_minimize(Context *context) {
                         return log_error_errno(errno, "Failed to open temporary file %s: %m", temp);
 
                 if (fstat(fd, &st) < 0)
-                        return log_error_errno(r, "Failed to stat temporary file: %m");
+                        return log_error_errno(errno, "Failed to stat temporary file: %m");
```
Comment (yuwata) [inline src/partition/repart.c:6362]: This is also a regression in v255 (5eef7047349).

File: src/partition/repart.c [hunk 3]
```diff
@@ -7569,7 +7569,7 @@ static int run(int argc, char *argv[]) {
 
                 r = search_and_access(d, F_OK, arg_root, CONF_PATHS_USR_STRV("systemd/repart/definitions"), &dp);
                 if (r < 0)
-                        return log_error_errno(errno, "DDI type '%s' is not defined: %m", arg_make_ddi);
+                        return log_error_errno(r, "DDI type '%s' is not defined: %m", arg_make_ddi);
```
Comment (yuwata) [inline src/partition/repart.c:7572]: This is also a regression in v255 (a121b331e37).

File: src/partition/repart.c [hunk 4]
```diff
@@ -4547,7 +4547,7 @@ static int do_copy_files(Context *context, Partition *p, const char *root) {
 
                 rfd = open(root, O_DIRECTORY|O_CLOEXEC|O_NOFOLLOW);
                 if (rfd < 0)
-                        return rfd;
+                        return -errno;
```
Comment (yuwata) [inline src/partition/repart.c:4550]: This is since v254 (3e451460211).

File: src/partition/repart.c [hunk 5]
```diff
@@ -3711,9 +3711,8 @@ static int partition_encrypt(Context *context, Partition *p, PartitionTarget *ta
 
                 /* Weird cryptsetup requirement which requires the header file to be the size of at least one
                  * sector. */
-                r = ftruncate(fileno(h), luks_params.sector_size);
-                if (r < 0)
-                        return log_error_errno(r, "Failed to grow temporary LUKS header file: %m");
+                if (ftruncate(fileno(h), luks_params.sector_size) < 0)
+                        return log_error_errno(errno, "Failed to grow temporary LUKS header file: %m");
```
Comment (yuwata) [inline src/partition/repart.c:3715]: This is since 983154f51831d0ee96ee08aed3367ea9990009e9 (v253).

**Review by yuwata — APPROVED** [review]
So, half of them are since v254. I think this should be in v255.

**Comment by poettering** [comment]
> Is this a regression in 255?

even if some of them are older, these are obvious bugs and low-risk bug fixes.

**Pull request merged**